lib/payment_dupes.rb

```diff
@@ -134,24 +170,23 @@ def self.remove_payment_dupes(np_id, designations_to_become_comments)
                                     if temp_offsite_matches.any?
                                         # it's the same donation for all of them so
                                         # we can do the copies once
-                                        copy_comment(offsite, online)
+                                        copy_comment(offsite, online, designations_to_become_comments)
                                         copy_dedication(offsite, online)
                                         copy_designation(offsite, online, designations_to_become_comments)
                                         deleted_payments.concat(temp_duplicate_payments)
                                         # deletes matching offsites here
-                                        p = Payment.find(temp_offsite_matches.first)
-                                        d = p.donation
-                                        d.payments.where(kind: 'OffsitePayment').each do |op|
```

Comment:
This would not work because an offsite payment would not belong to a donation that had multiple other payments.

Reply:
When you say it "would not work", what do you mean?

Reply:
It means it didn't work to delete all the offsite payments because a donation doesn't have multiple offsite payments.